### PR TITLE
1174 survey colours

### PR DESCRIPTION
**Pull request by willdoran**
This pull request makes the following changes:
- Adds color selection to surveys
- Adds coloring to posts based on their survey
- Adds marker icons and color to map

Test these changes by:
- Ensure that you can add/edit color for surveys
- Ensure that color appears in appropriate places based on the PL:
Specifically:
[map](http://github.ushahidi.org/platform-pattern-library/assets/html/5_layouts/map.html)
[timeline](http://github.ushahidi.org/platform-pattern-library/assets/html/5_layouts/timeline.htmll)
[post add/edit](http://github.ushahidi.org/platform-pattern-library/assets/html/5_layouts/post-add.html)
[post detail](http://github.ushahidi.org/platform-pattern-library/assets/html/5_layouts/post-detail.html)

Fixes ushahidi/platform#1174

Related client [273](https://github.com/ushahidi/platform-client/pull/273)

Ping @ushahidi/platform

---
This change is [Reviewable](https://reviewable.io/reviews/ushahidi/platform/1277)

### CODE REVIEW

File: src/Core/Entity/Form.php [hunk 1]
```diff
@@ -27,11 +28,17 @@ class Form extends StaticEntity
...
+			if ($color) {
+				return ltrim($color, '#');
+			}
+		};
 		return [
 			'id'          => 'int',
 			'parent_id'   => 'int',
 			'name'        => 'string',
 			'description' => 'string',
+			'color'				=> $typeColor,
```
Comment (rjmackay) [inline src/Core/Entity/Form.php]: should have spaces not tabs after `color`

File: application/classes/Ushahidi/Formatter/Form.php [hunk 2]
```diff
@@ -15,8 +15,15 @@ class Ushahidi_Formatter_Form extends Ushahidi_Formatter_API
 {
 	use FormatterAuthorizerMetadata;
 
-	protected function format_disabled($value) 
+	protected function format_disabled($value)
 	{
 		return (bool) $value;
 	}
+
+	protected function format_color($value)
```
Comment (rjmackay) [inline application/classes/Ushahidi/Formatter/Form.php]: adding basic tests. Will push :)

File: application/classes/Ushahidi/Formatter/Post/GeoJSON.php [hunk 3]
```diff
@@ -32,6 +32,7 @@ public function __invoke($entity)
 						'properties' => [
 							'title' => $entity->title,
 							'description' => $entity->content,
+							'color' => $entity->color,
```
Comment (rjmackay) [inline application/classes/Ushahidi/Formatter/Post/GeoJSON.php]: Can we use `marker-color` instead? (as in comments below) Probably pick a sane default for `marker-symbol` too

File: application/classes/Ushahidi/Formatter/Post/GeoJSONCollection.php [hunk 4]
```diff
@@ -54,6 +54,7 @@ public function __invoke($entities)
 					'properties' => [
 						'title' => $entity->title,
 						'description' => $entity->content,
+						'color' => $entity->color,
```
Comment (rjmackay) [inline application/classes/Ushahidi/Formatter/Post/GeoJSONCollection.php]: Same here - use marker-color

File: src/Core/Entity/Post.php [hunk 5]
```diff
@@ -22,6 +22,7 @@ class Post extends StaticEntity
 	protected $form_id;
 	protected $user_id;
 	protected $message_id;
+	protected $color;
```
Comment (rjmackay) [inline src/Core/Entity/Post.php]: Needs a comment to be clear this comes from elsewhere.

File: application/classes/Ushahidi/Formatter/Post/GeoJSON.php (hunk 3 above)
Comment (rjmackay) [inline application/classes/Ushahidi/Formatter/Post/GeoJSON.php]: ALso make sure to append `#` to the color here.

**Comment by rjmackay** [comment]
Reverting adding a spec test - it broke things but only on travis. Not worth the time right now.

**Comment by rjmackay** [comment]
@willdoran looks like you merged this despite changes requested from review??

**Comment by willdoran** [comment]
hmmm, that's strange i made the changes from review, let me check why they didn't commit

**Pull request closed without merging**